gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -56,19 +56,22 @@
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.ThreadSafe;
 
-/** Google Cloud Storage {@link FileSystemProvider} */
+/**
+ * Google Cloud Storage {@link FileSystemProvider}.
+ */
 @ThreadSafe
 @AutoService(FileSystemProvider.class)
 public final class CloudStorageFileSystemProvider extends FileSystemProvider {
 
-  private static final Supplier<GcsService> gcsServiceSupplier =
-      memoize(new Supplier<GcsService>() {
-        @Override
-        public GcsService get() {
-          return GcsServiceFactory.createGcsService();
-        }});
+  private final Storage gcsStorage;
```

Comment:
Nit, I would call it storage or gcs.

gcs stands for "google cloud storage", so adding a "Storage" suffix to it is a bit strange.